### PR TITLE
[WIP] Fixed #35. Updated CI to support multiple Ubuntu LTS releases

### LINKED ISSUE #35: Expand CI to all LTS Ubuntu releases
20.04 is already covered, but should add 18.04 and 16.04.

### PR DESCRIPTION
**Pull request by lextm**
* Added 18.04 and 16.04 support.
* Switched to container based images, so that later can add other Linux distributions (GitHub Actions hosts only Ubuntu).
* Revised the steps to adapt to 18.04/16.04 limitations (old Git, apt update errors and so on).

### CODE REVIEW

**Comment by knocte** [comment]
Latest commit has a red CI.

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ MSBUILD_LIBRARIES=Microsoft.Build.dll Microsoft.Build.Framework.dll Microsoft.Bu
 MSBUILD_DLLS=$(patsubst %, $(MSBUILD_PATH)/%, $(MSBUILD_LIBRARIES))
 
 # Set $PATH to point to provisioned .NET Core and avoid the ones provisioned by VSTS itself
-all: export PATH:="/usr/local/share/dotnet:$(PATH)"
+all: export PATH="/usr/local/share/dotnet:$PATH"
```
Comment (knocte) [inline Makefile]: why is this change needed? the commit message doesn't say the reason for it

File: .github/workflows/monodevelop.yml [hunk 2]
```diff
@@ -4,20 +4,138 @@ on:
...
+        apt-get install -y sudo        
+        sudo apt-get update
+        sudo apt-get install -y wget
+        wget https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt-get install -y apt-transport-https
+        sudo apt-get update
+        sudo apt-get install -y dotnet-sdk-3.1
+        sudo apt-get install -y gnupg ca-certificates
+        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
```
Comment (knocte) [inline .github/workflows/monodevelop.yml:91]: @lextm this script fragment is triplicated, should we create a sh script to avoid some duplication?

Comment (lextm) [reply .github/workflows/monodevelop.yml:91]: @knocte that's a good idea. Let's use this pull request to collect feedback only. I plan to open a new pull request later with some updates.

File: Makefile (hunk 1 above)
Comment (lextm) [reply Makefile]: @knocte The makefile change was made to let Ubuntu 16.04/18.04 compilation pass. If reverted, then those two platforms fail (https://github.com/lextm/dotdevelop/actions/runs/462099248).

Comment (knocte) [reply Makefile]: then add this rationale in the commit message pls

**Comment by lextm** [comment]
Will create a new pull request for this.
